monai/data/dataset.py

```diff
@@ -316,13 +315,15 @@ def _pre_transform(self, item_transformed):
             random transform object
 
         """
-        for _transform in self.transform.transforms:
-            # execute all the deterministic transforms
-            if isinstance(_transform, RandomizableTrait) or not isinstance(_transform, Transform):
-                break
-            # this is to be consistent with CacheDataset even though it's not in a multi-thread situation.
-            _xform = deepcopy(_transform) if isinstance(_transform, ThreadUnsafe) else _transform
-            item_transformed = apply_transform(_xform, item_transformed)
+        if not isinstance(self.transform, Compose):
+            raise ValueError("transform must be an instance of monai.transforms.Compose.")
```

Comment:
please make sure the PR can pass basic dataset tests:
```
python -m tests.test_cachedataset
python -m tests.test_persistentdatset
python -m tests.test_cachedataset_persistent_workers
```

Reply:
Done